packages/edit-site/src/components/dataviews/view-list.js

```diff
@@ -249,6 +253,54 @@ function ViewList( {
 			}
 			return column;
 		} );
+		if ( selection !== undefined ) {
+			_columns.unshift( {
```

Comment:
Is there any reason why you choose this approach instead of more simple one, using react table api?
I mean the same code here can be something like this:

```
	_columns.unshift( {
		header: ( props ) => {
			return (
				<CheckboxControl
					__nextHasNoMarginBottom
					checked={ props.table.getIsAllRowsSelected() }
					indeterminate={ props.table.getIsSomeRowsSelected() }
					onChange={ () =>
						props.table.toggleAllRowsSelected()
					}
				/>
			);
		},
		id: 'selection',
		cell: ( props ) => {
			return (
				<CheckboxControl
					__nextHasNoMarginBottom
					checked={ props.row.getIsSelected() }
					onChange={ props.row.getToggleSelectedHandler() }
				/>
			);
		},
		enableHiding: false,
		width: 40,
	} );
```

Reply:
Hi @Tropicalista,

Thank you for sharing insights and the code that uses the react table selection API. However, we have decided not to use it for two reasons:

Firstly, we have other views that are not compatible with tanstack, such as the grid view, and we need the selection to work seamlessly across all views.

Secondly, we are not certain about our future plans regarding tanstack, so we prefer to avoid using many of their API's and keep our options open.